File: src/tmp/Java/src/com/dbc/algorithm/Array/FlippingAnImage.java
package com.dbc.algorithm.Array;

public class FlippingAnImage {
    public static int[][] flipAndInvertImage(int[][] A) {
        for (int i = 0; i < A.length; i++) {
            int left = 0, right = A[0].length - 1;
            while (left <= right) {
                if (left == right) {
                    A[i][left] ^= 1;
                } else if (A[i][left] == A[i][right]) {
                    A[i][left] ^= 1;
                    A[i][right] ^= 1;
                }
                left++;
                right--;
            }
        }
        return A;
    }
}

File: core/tmp/Python/array/flipping_an_image.py
from typing import List

def flipAndInvertImage(A: List[List[int]]) -> List[List[int]]:
    row, col = len(A), len(A[0])
    for i in range(row):
        left, right = 0, col- 1
        while left <= right:
            if left == right:
                A[i][left] ^= 1
            elif A[i][left] == A[i][right]:
                A[i][left] ^= 1
                A[i][right] ^= 1
            left += 1
            right -= 1
    
    return A

if __name__ == "__main__":
    print(flipAndInvertImage([[1,1,0],[1,0,1],[0,0,0]]))